packages/kbn-shared-ux-components/src/page_template/solution_nav/solution_nav.tsx

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import './solution_nav.scss';
+
+import React, { useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import {
+  EuiFlyout,
+  EuiSideNav,
+  EuiSideNavItemType,
+  EuiSideNavProps,
+  useIsWithinBreakpoints,
+} from '@elastic/eui';
+
+import classNames from 'classnames';
+import {
+  KibanaPageTemplateSolutionNavAvatar,
+  KibanaPageTemplateSolutionNavAvatarProps,
+} from './solution_nav_avatar';
+import { KibanaPageTemplateSolutionNavCollapseButton } from './solution_nav_collapse_button';
+
+export type KibanaPageTemplateSolutionNavProps = EuiSideNavProps<{}> & {
+  /**
+   * Name of the solution, i.e. "Observability"
+   */
+  name: KibanaPageTemplateSolutionNavAvatarProps['name'];
+  /**
+   * Solution logo, i.e. "logoObservability"
+   */
+  icon?: KibanaPageTemplateSolutionNavAvatarProps['iconType'];
+  /**
+   * Control the collapsed state
+   */
+  isOpenOnDesktop?: boolean;
+  onCollapse?: () => void;
+};
+
+const SIZE = 248;
+
+const setTabIndex = (items: Array<EuiSideNavItemType<{}>>, isHidden: boolean) => {
+  return items.map((item) => {
+    // @ts-ignore-next-line Can be removed on close of https://github.com/elastic/eui/issues/4925
+    item.tabIndex = isHidden ? -1 : undefined;
```

Comment:
Looks like the issue has been closed, but there is still no `tabIndex` prop on `EuiSideNavItemType`. Do we need this?

Reply:
It's an accessiblity issue I'd have to dig into. I'd leave for now